src/sisl/geom/_neighbors/_finder.py

```diff
@@ -3,17 +3,213 @@
 # file, You can obtain one at https://mozilla.org/MPL/2.0/.
 from __future__ import annotations
 
-from typing import Optional, Sequence, Tuple, Union
+from functools import cached_property
+from typing import Literal, Optional, Sequence, Tuple, Union
 
 import numpy as np
 
-from sisl import Geometry
+from sisl import Geometry, Lattice
 from sisl.typing import AtomsIndex
 from sisl.utils import size_to_elements
 
 from . import _operations
 
 
+class Interactions:
+
+    def __init__(self, geometry: Geometry, finder_results, split_indices=None):
+        self.geometry = geometry
+        self.finder_results = finder_results
+        self._split_indices = split_indices
+
+    @property
+    def i(self):
+        return self.finder_results[:, 0]
+
+    @property
+    def j(self):
+        return self.finder_results[:, 1]
+
+    @property
+    def isc(self):
+        return self.finder_results[:, 2:]
+
+    @cached_property
+    def nsc(self):
+        return np.max(np.abs(self.isc), axis=0) * 2 + 1
+
+    @property
+    def j_sc(self):
+        lattice = Lattice(cell=[[1, 0, 0], [0, 1, 0], [0, 0, 1]], nsc=self.nsc)
+
+        return self.j + lattice.sc_index(self.isc) * self.geometry.na
+
+
+class NeighborsList(Interactions):
+    """Base class for interactions between atoms."""
+
+    @cached_property
+    def n_interactions(self):
+        """Number of interactions each atom participates in."""
+
+        if self._split_indices is None:
+            n_interactions = np.zeros(self.geometry.na, dtype=int)
+            index, counts = np.unique(self.i, return_counts=True)
+
+            n_interactions[index] = counts
+            return n_interactions
+        else:
+            return np.diff(self._split_indices, prepend=0)
+
+
+class UniqueNeighborsList(NeighborsList):
+    """Full neighbors list of a system, but containing only the upper triangle of the adjacency matrix.
+
+    What this means, is that the the object only contains one direction of each interaction.
+    This is only possible if the interaction is symmetric (there is no directionality and
+    thresholds for interaction do not depend on direction).
+    """
+
+    @cached_property
+    def n_interactions(self):
```

Comment:
I can understand the wording here, but could we use something else, e.g. `n_neigbors`? It also goes in spirit of the class names.

Reply:
I wasn't sure if `neighbors` would be general enough for the case where one looks for atoms close to points in space.

Does it make sense to call the atoms "neighbors of a point"?

If so maybe we can change the base class name to `Neighbors`.

Reply:
Yes, I would say so :)
To both things :)